### ISSUE TITLE
Make it easier to share configs, run commands for finetuning

### ISSUE THREAD

**Issue by rohan-varma** [pytorch/torchtune#220]
- Oftentimes, teammates or other folks would like to reproduce the results that you've observed in your finetune (to replicate/confirm correctness or reproduce bugs for example). This issue is to brainstorm and formulate a good approach for doing so correctly and document the process one could follow to do this in torchtune.
- Critical items needed to reproduce results: exact config (including details like FSDP configuration, optimizer setting), seed value, # of GPUs used, device (not sure if I'm missing anything else)
